core/deployment/src/main/java/io/quarkus/deployment/steps/DevModeBuildStep.java

```diff
@@ -11,6 +12,7 @@
     @BuildStep
     List<HotDeploymentWatchedFileBuildItem> config() {
         return Arrays.asList(new HotDeploymentWatchedFileBuildItem("META-INF/microprofile-config.properties"),
-                new HotDeploymentWatchedFileBuildItem("application.properties"));
+                new HotDeploymentWatchedFileBuildItem("application.properties"), new HotDeploymentWatchedFileBuildItem(
+                        Paths.get("").toAbsolutePath().normalize().resolve(".env").toAbsolutePath().toString()));
```

Comment:
This seems like it might be overkill... would something like this work?

```suggestion
                        Paths.get(".env").toAbsolutePath().toString()));
```

Reply:
It does yes, thanks